mysql-replicator-streams/src/main/java/com/booking/replication/streams/StreamsImplementation.java

```diff
@@ -112,6 +117,11 @@
         this.handler = (exception) -> StreamsImplementation.LOG.error("error inside streams", exception);
     }
 
+    public void registerMetric(MetricRegistry metricRegistry) {
+        metricRegistry.removeMatching(MetricFilter.startsWith(METRIC_PREFIX));
+        metricRegistry.register(METRIC_PREFIX, new QueuesMetricSet<Input>(this.queues));
```

Comment:
Register metrics is already behind Metric abstraction. Use apis from Metric classes